core/factorization/lu.cpp

```diff
@@ -69,6 +69,9 @@ Lu<ValueType, IndexType>::parse(const config::pnode& config,
     if (auto& obj = config.get("skip_sorting")) {
         params.with_skip_sorting(config::get_value<bool>(obj));
     }
+    if (auto& obj = config.get("has_all_fillin")) {
+        params.with_has_all_fillin(config::get_value<bool>(obj));
```

Comment:
sounds good to me.
Do you think it will be a bit misleading for the case we were discussing, the factorization does not contains all values from the matrices?

Reply:
Also, based on the update comment in `lu.hpp` (the above coversation), a factorization that doesn't contain all the locations from the original matrix currently isn't allowed, anyway, right?

Reply:
right. The current version can not work with the factorization misses some entries from the original matrix. I just wonder whether should consider the situation when the factorization does not contains the nonzeros from the original matrix. To do so, we also need to adapt it in the `initialize` function. When we allow thissituation, we definitely do not want to introduce another option to enable it.

Reply:
Oh, I see -- just looking to the future. I don't think it's a problem, personally -- I think "fill-in" is pretty well-understood to mean "all the additional locations of the factorization vs original matrix" so I don't think there'd be any confusion that it's meant to differentiate between incomplete factorization with or without all the original matrix locations.